Sources/GRPC/AsyncAwaitSupport/GRPCAsyncServerHandler.swift

```diff
@@ -0,0 +1,566 @@
+/*
+ * Copyright 2021, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#if compiler(>=5.5)
+
+import _NIOConcurrency
+import NIOCore
+import NIOHPACK
+
+@available(macOS 12, iOS 15, tvOS 15, watchOS 8, *)
+public struct GRPCAsyncServerHandler<
```

Comment:
The underlying type _is_ a reference type. We only wrap this up as a `struct` because we don't want to expose the underlying type to public API (generated code will return instances of this) so that we can change the implementation later.